lib/parse.js

```diff
@@ -78,6 +81,8 @@ var parseValues = function parseQueryStringValues(str, options) {
             val = options.decoder(part.slice(pos + 1), defaults.decoder, charset);
         }
 
+        key = options.mapKeys(key);
```

Comment:
this should be extracted out so that the `mapKeys` function does not get `options` as its receiver (`this` value)